src/arrayMethodShift.js

```diff
@@ -5,7 +5,15 @@
  */
 function applyCustomShift() {
   [].__proto__.shift2 = function() {
-    // write code here
+    this.reverse();
+    let a = this[this.length - 1];
+    if (this.length === 0) {
```

Comment:
This should be before `this.reverse`. You don't need to do extra work if `this.length === 0`